src/Adapter/Module/Tab/ModuleTabRegister.php

```diff
@@ -275,15 +282,18 @@ protected function registerTab(Module $module, ParameterBag $tabDetails)
     {
         $this->checkIsValid($module->get('name'), $tabDetails);
 
-        /**
-         * Legacy Tab, to be replaced with Doctrine entity when right management
-         * won't be directly linked to the tab creation
-         *
-         * @ToDo
-         */
+        // Legacy Tab, to be replaced with Doctrine entity when right management
+        // won't be directly linked to the tab creation
+        // @ToDo
+
+        $className = $tabDetails->get('class_name');
+
+        if (is_subclass_of($className, FrameworkBundleAdminController::class)) {
+            $className = $tabDetails->get('legacy_class_name');
```

Comment:
thinking about it, it's a very naive implementation: I should check for the existence of this parameter and add a meaningful message pointing to the docs: wdyt?